### PR TITLE
Issues/380 fido2 config

### PR DESCRIPTION
**Pull request by jnpsk**
Added configuration classes for `PowerAuthClient` and `PowerAuthFido2Client`.  
The `WebAuthnConfiguration` holds WebAuthn properties, that are not configurable in UI.

### CODE REVIEW

File: powerauth-fido2-tests/lombok.config [hunk 1]
```diff
@@ -0,0 +1 @@
+lombok.log.fieldName=logger
```
Comment (banterCZ) [inline powerauth-fido2-tests/lombok.config:1]: 🤔 This is the second lombok config in this repo. May a single one in the root could be enough.

File: powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/PowerAuthFido2WebServiceConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            // Endpoint security is on
+            if (clientToken != null && !clientToken.isEmpty()) {
+                final PowerAuthRestClientConfiguration config = new PowerAuthRestClientConfiguration();
+                config.setPowerAuthClientToken(clientToken);
+                config.setPowerAuthClientSecret(clientSecret);
+                return new PowerAuthFido2RestClient(powerAuthServiceBaseUrl, config);
+            } else {
+                return new PowerAuthFido2RestClient(powerAuthServiceBaseUrl);
+            }
+        } catch (PowerAuthClientException ex) {
```
Comment (banterCZ) [inline powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/PowerAuthFido2WebServiceConfiguration.java:60]: I know, that this is a pattern in our codebase but do you want to really recover from this error? Spring IoC will fail with `null`, I guess. I suggest to declare `public PowerAuthFido2Client powerAuthFido2Client() throws PowerAuthClientException {`

File: powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/PowerAuthFido2WebServiceConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * PowerAuth FIDO2 service configuration class.
+ *
+ * @author Jan Pesek, jan.pesek@wultra.com
+ */
+@Configuration
+@Slf4j
+public class PowerAuthFido2WebServiceConfiguration {
+
+    @Value("${powerauth.service.baseUrl}")
```
Comment (banterCZ) [inline powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/PowerAuthFido2WebServiceConfiguration.java:39]: What about to introduce `@ConfigurationProperties`?

File: powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/PowerAuthWebServiceConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            // Endpoint security is on
+            if (clientToken != null && !clientToken.isEmpty()) {
+                final PowerAuthRestClientConfiguration config = new PowerAuthRestClientConfiguration();
+                config.setPowerAuthClientToken(clientToken);
+                config.setPowerAuthClientSecret(clientSecret);
+                return new PowerAuthRestClient(powerAuthServiceUrl, config);
+            } else {
+                return new PowerAuthRestClient(powerAuthServiceUrl);
+            }
+        } catch (PowerAuthClientException ex) {
```
Comment (banterCZ) [inline powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/PowerAuthWebServiceConfiguration.java:64]: dtto

File: powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/WebAuthnConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.context.annotation.Configuration;
+
+import java.util.List;
+
+/**
+ * WebAuthn configuration properties.
+ *
+ * @author Jan Pesek, jan.pesek@wultra.com
+ */
+@Configuration
```
Comment (banterCZ) [inline powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/WebAuthnConfiguration.java:32]: Why not `@ConfigurationProperties`?

File: powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/PowerAuthFido2WebServiceConfiguration.java (hunk 2 above)
Comment (jnpsk) [reply powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/PowerAuthFido2WebServiceConfiguration.java:60]: thats a good idea, thanks

File: powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/WebAuthnConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+@ConfigurationProperties(prefix = "powerauth.webauthn")
+@Data
+public class WebAuthnConfiguration {
+
+    private String rpId;
+    private String rpName;
+    private Long timeout;
```
Comment (banterCZ) [inline powerauth-fido2-tests/src/main/java/com/wultra/security/powerauth/fido2/configuration/WebAuthnConfiguration.java:39]: May we use `java.util.Duration`?

**Review by banterCZ — APPROVED** [review]
Looks good now.

**Pull request merged**